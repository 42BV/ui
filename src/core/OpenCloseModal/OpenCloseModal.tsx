import React, { FormEventHandler, Suspense } from 'react';
import { Button } from '../Button/Button';
import { SubmitButton } from '../SubmitButton/SubmitButton';
import { t } from '../../utilities/translation/translation';
import { BootstrapSize } from '../types';
import { IconType } from '../Icon';
import { Modal, ModalBody, ModalFooter, ModalHeader } from 'reactstrap';
import { Loading } from '../Loading/Loading';

type Text = {
  cancel?: string;
  save?: string;
};

type Props = {
  /**
   * Whether the save action you are performing is currently in
   * progress. If so a spinner is rendered inside the button.
   * This behavior is optional and default to `false`.
   */
  inProgress?: boolean;

  /**
   * Callback for when the modal should close.
   */
  onClose: () => void;

  /**
   * Optionally the icon of the save button.
   *
   * Defaults to `save`
   */
  saveIcon?: IconType;

  /**
   * Optionally icon of the cancel button.
   *
   * Defaults to `cancel`
   */
  cancelIcon?: IconType;

  /**
   * Callback for when the save button is clicked.
   */
  onSave?: FormEventHandler;

  /**
   * The content of the modal body.
   */
  children: React.ReactNode;

  /**
   * The label of the form element.
   */
  label?: React.ReactNode;

  /**
   * Optionally customized text within the component.
   * This text should already be translated.
   */
  text?: Text;

  /**
   * Optionally the size (width) of the modal.
   */
  size?: BootstrapSize;

  /**
   * Whether the footer should stick to the bottom of the modal.
   * This allows the user to always click the close and save buttons.
   *
   * Defaults to `true`.
   */
  stickyFooter?: boolean;

  /**
   * Optional extra CSS class you want to add to the component.
   * Useful for styling the component.
   */
  className?: string;

  /**
   * Optional extra CSS class you want to add to the <ModalHeader>.
   * Useful for styling the component.
   */
  modalHeaderClassName?: string;

  /**
   * Optional extra CSS class you want to add to the <ModalBody>.
   * Useful for styling the component.
   */
  modalBodyClassName?: string;

  /**
   * Optional extra CSS class you want to add to the <ModalFooter>.
   * Useful for styling the component.
   */
  modalFooterClassName?: string;
};

/**
 * Often a modal is content with a header and a cancel and save button
 * in the footer. To make it easier for you developers, we included the
 * OpenCloseModal which already includes the cancel and save button to
 * save you some code duplication for all those similar modals. If you
 * want a more advanced display, you should use the `<Modal>` component.
 */
export function OpenCloseModal(props: Props) {
  const {
    inProgress,
    onClose,
    onSave,
    children,
    label,
    text,
    size,
    className,
    saveIcon = 'save',
    cancelIcon = 'cancel',
    stickyFooter = true,
    modalHeaderClassName,
    modalBodyClassName,
    modalFooterClassName
  } = props;

  const header = label ? (
    <ModalHeader toggle={onClose} className={modalHeaderClassName}>
      {label}
    </ModalHeader>
  ) : null;

  const body = (
    <ModalBody className={modalBodyClassName}>
      <Suspense fallback={<Loading />}>{children}</Suspense>
    </ModalBody>
  );

  return (
    <Modal
      toggle={onClose}
      className={className}
      isOpen={true}
      size={size}
      wrapClassName={stickyFooter ? 'sticky-modal' : undefined}
    >
      {onSave ? (
        <form onSubmit={onSave}>
          {header}
          {body}
          <ModalFooter className={modalFooterClassName}>
            <Button
              className="ms-1"
              color="secondary"
              icon={cancelIcon}
              onClick={onClose}
            >
              {t({
                overrideText: text?.cancel,
                key: 'OpenCloseModal.CANCEL',
                fallback: 'Cancel'
              })}
            </Button>
            <SubmitButton
              className="ms-1"
              inProgress={!!inProgress}
              icon={saveIcon}
            >
              {t({
                overrideText: text?.save,
                key: 'OpenCloseModal.SAVE',
                fallback: 'Save'
              })}
            </SubmitButton>
          </ModalFooter>
        </form>
      ) : (
        <>
          {header}
          {body}
        </>
      )}
    </Modal>
  );
}
