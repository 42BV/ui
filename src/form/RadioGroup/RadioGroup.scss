.radio-group > legend {
  font-size: 1rem;
}
