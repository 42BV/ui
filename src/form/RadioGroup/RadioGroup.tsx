import React from 'react';
import { FormGroup, Input, Label } from 'reactstrap';
import { constant, get } from 'lodash';

import withJarb from '../withJarb/withJarb';
import { Color } from '../types';
import { t } from '../../utilities/translation/translation';
import {
  isOptionSelected,
  OptionEnabledCallback,
  OptionEqual,
  OptionForValue,
  OptionsFetcher
} from '../option';
import { doBlur } from '../utils';
import Loading from '../../core/Loading/Loading';
import { useOptions } from '../useOptions';

export interface Text {
  /**
   * The message to show when the select is loading. Defaults
   * to `loading...`
   */
  loadingMessage?: string;
}

interface BaseProps<T> {
  /**
   * The id of the form element.
   */
  id?: string;

  /**
   * The value that the form element currently has.
   */
  value?: T;

  /**
   * Callback for when the form element changes.
   */
  onChange: (value: T) => void;

  /**
   * Optional callback for when the form element is blurred.
   */
  onBlur?: () => void;

  /**
   * Is either an array of options or a callback which fetches
   * the options asynchronously.
   */
  options: OptionsFetcher<T> | T[];

  /**
   * Callback to convert an value of type T to an option to show
   * to the user.
   */
  optionForValue: OptionForValue<T>;

  /**
   * Optional callback which is used to determine if two options
   * of type T are equal.
   *
   * When `isOptionEqual` is not defined the outcome of `optionForValue`
   * is used to test equality.
   */
  isOptionEqual?: OptionEqual<T>;

  /**
   * Optional callback which is called for every option to determine
   * if the option can be selected. By default all options can be
   * selected.
   */
  isOptionEnabled?: OptionEnabledCallback<T>;

  /**
   * Optionally the error message to render.
   */
  error?: React.ReactNode;

  /**
   * Optionally the color of the FormGroup.
   */
  color?: Color;

  /**
   * Whether or not the form element is currently valid.
   */
  valid?: boolean;

  /**
   * Optional extra CSS class you want to add to the component.
   * Useful for styling the component.
   */
  className?: string;

  /**
   * The placeholder of the form element.
   */
  placeholder?: string;

  /**
   * Optionally customized text within the component.
   * This text should already be translated.
   */
  text?: Text;
}

interface WithoutLabel<T> extends BaseProps<T> {
  label?: never;
}

interface WithLabel<T> extends BaseProps<T> {
  /**
   * The label of the form element.
   */
  label: string;
}

export type Props<T> = WithoutLabel<T> | WithLabel<T>;

/**
 * RadioGroup is a form element for which the value can be selected
 * from a limited range.
 */
export default function RadioGroup<T>(props: Props<T>) {
  const {
    value,
    error,
    color,
    text = {},
    className = '',
    placeholder,
    onChange,
    onBlur,
    optionForValue,
    isOptionEqual
  } = props;

  const { options, loading } = useOptions({
    optionsOrFetcher: props.options,
    value,
    isOptionEqual,
    optionForValue
  });

  const isOptionEnabled = get(props, 'isOptionEnabled', constant(true));

  function onRadioClicked(option: T) {
    onChange(option);
    doBlur(onBlur);
  }

  return (
    <FormGroup tag="fieldset" className={className} color={color}>
      {'label' in props && props.label ? <legend>{props.label}</legend> : null}
      {placeholder ? (
        <p className="text-muted">
          <em>{placeholder}</em>
        </p>
      ) : null}
      {loading ? (
        <Loading>
          {t({
            key: 'RadioGroup.LOADING',
            fallback: 'Loading...',
            overrideText: text.loadingMessage
          })}
        </Loading>
      ) : (
        options.map(option => {
          const label = optionForValue(option);

          return (
            <FormGroup key={label} check>
              <Label check>
                <Input
                  type="radio"
                  value={label}
                  checked={isOptionSelected({
                    option,
                    optionForValue,
                    isOptionEqual,
                    value
                  })}
                  disabled={!isOptionEnabled(option)}
                  onChange={() => onRadioClicked(option)}
                />{' '}
                {label}
              </Label>
            </FormGroup>
          );
        })
      )}

      {error}
    </FormGroup>
  );
}
/**
 * Variant of the RadioGroup which can be used in a Jarb context.
 */
export const JarbRadioGroup = withJarb<any, any, Props<any>>(RadioGroup);
