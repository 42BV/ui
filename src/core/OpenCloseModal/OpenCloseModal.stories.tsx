import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { OpenCloseModal } from './OpenCloseModal';
import { Card } from 'reactstrap';
import Button from '../Button/Button';
import RadioGroup from '../../form/RadioGroup/RadioGroup';

storiesOf('core|OpenCloseModal', module)
  .add('basic', () => {
    const [isOpen, setIsOpen] = useState(false);
    const [inProgress, setInProgress] = useState(false);
    const [choice, setChoice] = useState();

    function onSave() {
      action('save button was clicked');

      setInProgress(true);

      setTimeout(() => {
        setIsOpen(false);
      }, 2000);
    }

    function onClose() {
      action('modal was closed');
      setIsOpen(false);
    }

    return (
      <Card body>
        <Button onClick={() => setIsOpen(true)}>Start inquiry</Button>
        <OpenCloseModal
          isOpen={isOpen}
          inProgress={inProgress}
          onClose={onClose}
          onSave={onSave}
          label="What are you doing?"
        >
          <RadioGroup
            value={choice}
            onChange={setChoice}
            options={[
              'Watching a movie',
              'I work at 42, the most awesome company in the world!',
              'Nothing'
            ]}
            optionForValue={option => option}
          />
        </OpenCloseModal>
      </Card>
    );
  })
  .add('without heading', () => {
    const [isOpen, setIsOpen] = useState(false);

    return (
      <Card body>
        <Button onClick={() => setIsOpen(true)}>Inspire me</Button>
        <OpenCloseModal
          isOpen={isOpen}
          onSave={() => setIsOpen(false)}
          onClose={() => setIsOpen(false)}
          text={{ save: 'Yes', cancel: 'No' }}
        >
          <p>
            You have to take a break once in a while. Did you go for a walk
            outside to breathe some fresh air?
          </p>
        </OpenCloseModal>
      </Card>
    );
  })
  .add('without buttons', () => {
    const [isOpen, setIsOpen] = useState(false);

    function onClose() {
      setIsOpen(false);
    }

    return (
      <Card body>
        <Button onClick={() => setIsOpen(true)}>What does Baymax say?</Button>
        <OpenCloseModal
          isOpen={isOpen}
          onClose={onClose}
          label="Your personal health companion"
        >
          <p>
            You need to wash your hands if you want to prevent getting Corona
            virus.
          </p>
        </OpenCloseModal>
      </Card>
    );
  })
  .add('custom button text', () => {
    const [isOpen, setIsOpen] = useState(false);
    const [choice, setChoice] = useState();

    function onSave() {
      alert(`You chose to ${choice.value} the diamond`);
      setIsOpen(false);
    }

    function onClose() {
      alert(
        `Darn! You tricked and the diamond fell in an endless hole in the ground...`
      );
      setIsOpen(false);
    }

    return (
      <Card body>
        <p>You found a box!</p>
        <Button onClick={() => setIsOpen(true)}>
          Check what&apos;s inside
        </Button>
        <OpenCloseModal
          isOpen={isOpen}
          onClose={onClose}
          onSave={onSave}
          label="It's a special gift!"
          text={{
            save: 'Confirm',
            cancel: ''
          }}
        >
          <p>Inside the box you find a diamond. What do you want to do?</p>
          <RadioGroup
            value={choice}
            onChange={setChoice}
            options={[
              { label: 'Sell it!', value: 'sell' },
              { label: 'Put it in my pocket', value: 'store' }
            ]}
            optionForValue={option => option.label}
          />
        </OpenCloseModal>
      </Card>
    );
  });
