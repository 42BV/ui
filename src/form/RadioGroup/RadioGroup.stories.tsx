import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';

import { JarbRadioGroup, RadioGroup } from './RadioGroup';
import {
  FinalForm,
  IsOptionEqualInfo,
  JarbFormElementDependencies,
  KeyForOptionInfo,
  nonExistingProvince,
  Province,
  provinceFetcher,
  provinces,
  ReloadOptionsInfo,
  resolveAfter
} from '../story-utils';
import { Card } from '../../core/Card/Card';
import { pageOf } from '../../utilities/page/page';
import { Tooltip } from '../../core/Tooltip/Tooltip';
import { Icon } from '../../core/Icon';

storiesOf('Form/RadioGroup', module)
  .addParameters({ component: RadioGroup })
  .add('predefined options', () => {
    const [ value, setValue ] = useState<Province | undefined>(
      nonExistingProvince()
    );

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}
      </Card>
    );
  })
  .add('async options', () => {
    const [ value, setValue ] = useState<Province | undefined>(provinces()[0]);

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinceFetcher}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}

        <p className="mt-3">
          <strong>Disclaimer:</strong> when using async, a maximum of 100
          options will be displayed. If you want to display more than 100
          options, you should use the ModalPickerSingle.
        </p>
      </Card>
    );
  })
  .add('disabled options', () => {
    const [ value, setValue ] = useState<Province | undefined>(
      nonExistingProvince()
    );

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          isOptionEnabled={(option) => !option.label.startsWith('Z')}
          value={value}
          onChange={setValue}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}
      </Card>
    );
  })
  .add('custom isOptionEqual', () => {
    const [ value, setValue ] = useState<Province | undefined>(provinces()[0]);

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          isOptionEqual={(a, b) => a.value === b.value}
          value={value}
          onChange={setValue}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}

        <IsOptionEqualInfo />
      </Card>
    );
  })
  .add('custom keyForOption', () => {
    const [ value, setValue ] = useState<Province | undefined>(provinces()[0]);

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          keyForOption={(province) => province.value}
          value={value}
          onChange={setValue}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}

        <KeyForOptionInfo />
      </Card>
    );
  })
  .add('using reloadOptions', () => {
    const [ brand, setBrand ] = useState<string>();
    const [ model, setModel ] = useState<string>();

    const allOptions = {
      Audi: [ 'A1', 'A2', 'A3', 'M5' ],
      BMW: [ 'series 1', 'series 2', 'series 3', 'series 4', 'series 5' ],
      Mercedes: [ 'Viano', 'Vito', 'Sprinter' ]
    };

    return (
      <Card className="m-2">
        <RadioGroup
          id="brand"
          label="Brand"
          placeholder="Please select your brand"
          options={() => resolveAfter(pageOf(Object.keys(allOptions), 1))}
          labelForOption={(option) => option}
          value={brand}
          onChange={(value) => {
            setBrand(value);
            setModel(undefined);
          }}
        />
        <RadioGroup
          id="model"
          label="Model"
          placeholder={
            brand ? 'Please select your model' : 'Please select a brand first'
          }
          options={() =>
            resolveAfter(pageOf(brand ? allOptions[brand] : [], 1))
          }
          labelForOption={(option: string) => option}
          value={model}
          onChange={setModel}
          reloadOptions={brand}
        />

        {brand ? <p>Your chosen brand is: {brand}</p> : null}
        {model ? <p>Your chosen model is: {model}</p> : null}

        <ReloadOptionsInfo />
      </Card>
    );
  })
  .add('label & placeholder', () => {
    const [ value, setValue ] = useState<Province | undefined>(
      nonExistingProvince()
    );

    return (
      <Card className="m-2">
        <span className="d-block fs-5">Invisible label</span>

        <RadioGroup<Province>
          label="Province"
          hiddenLabel={true}
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
        />

        <hr />

        <span className="d-block fs-5">Custom label</span>

        <RadioGroup<Province>
          id="provinces"
          label={
            <div className="d-flex justify-content-between">
              <span>Subject</span>
              <Tooltip className="ms-1" content="The province you live in">
                <Icon icon="info" />
              </Tooltip>
            </div>
          }
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
        />

        <hr />

        <span className="d-block fs-5">Without placeholder</span>

        <RadioGroup<Province>
          id="province"
          label="Province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
        />

        <hr />

        {value ? <p>Your chosen province is: {value.label}</p> : null}
      </Card>
    );
  })
  .add('horizontal', () => {
    const [ value, setValue ] = useState<Province | undefined>(
      nonExistingProvince()
    );

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
          horizontal={true}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}

        <p>
          <strong>Disclaimer:</strong> horizontal mode works best when there are
          not too many items
        </p>
      </Card>
    );
  })
  .add('horizontal with clear', () => {
    const [ value, setValue ] = useState<Province | undefined>(
      nonExistingProvince()
    );

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
          horizontal={true}
          canClear={true}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}

        <p>
          <strong>Disclaimer:</strong> horizontal mode works best when there are
          not too many items
        </p>
      </Card>
    );
  })
  .add('with clear button', () => {
    const [ value, setValue ] = useState<Province | undefined>(
      nonExistingProvince()
    );

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
          canClear={true}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}
      </Card>
    );
  })

  .add('jarb', () => {
    return (
      <>
        <JarbFormElementDependencies />
        <FinalForm>
          <JarbRadioGroup
            id="province"
            name="province"
            label="Province"
            placeholder="Please select your province"
            options={provinceFetcher}
            labelForOption={(province) => province.label}
            jarb={{
              validator: 'User.province',
              label: 'Province'
            }}
          />
        </FinalForm>
      </>
    );
  });
