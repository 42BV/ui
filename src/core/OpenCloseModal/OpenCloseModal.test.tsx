import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import RadioGroup from '../../form/RadioGroup/RadioGroup';

import { OpenCloseModal } from './OpenCloseModal';

describe('Component: OpenCloseModal', () => {
  let openCloseModal: ShallowWrapper;
  let onCloseSpy: jest.Mock;
  let onSaveSpy: jest.Mock;

  function setup({
    isOpen = false,
    hasLabel = true,
    hasCustomText = false,
    hasButtons = true,
    inProgress = undefined
  }: {
    isOpen?: boolean;
    hasLabel?: boolean;
    hasCustomText?: boolean;
    hasButtons?: boolean;
    inProgress?: boolean;
  }) {
    onCloseSpy = jest.fn();
    onSaveSpy = jest.fn();

    const props = {
      isOpen,
      inProgress,
      onClose: onCloseSpy,
      text: hasCustomText
        ? { cancel: 'Stop please', save: 'Select me' }
        : undefined
    };

    if (hasButtons) {
      // @ts-ignore
      props.onSave = onSaveSpy;
    }

    const children = (
      <RadioGroup
        onChange={jest.fn()}
        options={['local', 'development', 'test', 'acceptation', 'production']}
        optionForValue={v => v}
      />
    );

    if (hasLabel) {
      openCloseModal = shallow(
        <OpenCloseModal label="Choose something" {...props}>
          {children}
        </OpenCloseModal>
      );
    } else {
      openCloseModal = shallow(
        <OpenCloseModal {...props}>{children}</OpenCloseModal>
      );
    }
  }

  describe('ui', () => {
    test('with label', () => {
      setup({ isOpen: true });

      expect(toJson(openCloseModal)).toMatchSnapshot(
        'Component: OpenCloseModal => ui => with label'
      );
    });

    test('without label', () => {
      setup({ hasLabel: false });

      expect(toJson(openCloseModal)).toMatchSnapshot(
        'Component: OpenCloseModal => ui => without label'
      );
    });

    test('without buttons', () => {
      setup({ hasButtons: false });

      expect(toJson(openCloseModal)).toMatchSnapshot(
        'Component: OpenCloseModal => ui => without buttons'
      );
    });

    test('custom button texts', () => {
      setup({ hasCustomText: true });

      expect(toJson(openCloseModal)).toMatchSnapshot(
        'Component: OpenCloseModal => ui => custom button texts'
      );
    });

    test('in progress', () => {
      setup({ inProgress: true });

      expect(toJson(openCloseModal)).toMatchSnapshot(
        'Component: OpenCloseModal => ui => in progress'
      );
    });
  });

  describe('events', () => {
    it('should call onClose when clicked outside modal', () => {
      setup({});

      // @ts-ignore
      openCloseModal
        .find('Modal')
        .at(0)
        .props()
        // @ts-ignore
        .toggle();

      expect(onCloseSpy).toBeCalledTimes(1);
    });

    it('should call onClose when close button clicked', () => {
      setup({});

      // @ts-ignore
      openCloseModal
        .find('ModalHeader')
        .at(0)
        .props()
        // @ts-ignore
        .toggle();

      expect(onCloseSpy).toBeCalledTimes(1);
    });

    it('should call onClose when cancel button clicked', () => {
      setup({});

      // @ts-ignore
      openCloseModal
        .find('Button')
        .at(0)
        .props()
        // @ts-ignore
        .onClick();

      expect(onCloseSpy).toBeCalledTimes(1);
    });

    it('should call onSave when save button clicked', () => {
      setup({});

      // @ts-ignore
      openCloseModal
        .find('Button')
        .at(1)
        .props()
        // @ts-ignore
        .onClick();

      expect(onSaveSpy).toBeCalledTimes(1);
    });
  });
});
