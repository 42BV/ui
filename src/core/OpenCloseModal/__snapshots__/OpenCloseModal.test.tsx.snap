// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: OpenCloseModal ui custom button texts: Component: OpenCloseModal => ui => custom button texts 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  fade={true}
  isOpen={false}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  returnFocusAfterClose={true}
  role="dialog"
  scrollable={false}
  toggle={[Function]}
  unmountOnClose={true}
  zIndex={1050}
>
  <ModalHeader
    charCode={215}
    closeAriaLabel="Close"
    tag="h5"
    toggle={[Function]}
    wrapTag="div"
  >
    Choose something
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <RadioGroup
      onChange={[MockFunction]}
      optionForValue={[Function]}
      options={
        Array [
          "local",
          "development",
          "test",
          "acceptation",
          "production",
        ]
      }
    />
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <Button
      className="ml-1"
      color="secondary"
      onClick={[Function]}
    >
      Stop please
    </Button>
    <Button
      className="ml-1"
      color="primary"
      onClick={[Function]}
    >
      Select me
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`Component: OpenCloseModal ui in progress: Component: OpenCloseModal => ui => in progress 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  fade={true}
  isOpen={false}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  returnFocusAfterClose={true}
  role="dialog"
  scrollable={false}
  toggle={[Function]}
  unmountOnClose={true}
  zIndex={1050}
>
  <ModalHeader
    charCode={215}
    closeAriaLabel="Close"
    tag="h5"
    toggle={[Function]}
    wrapTag="div"
  >
    Choose something
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <RadioGroup
      onChange={[MockFunction]}
      optionForValue={[Function]}
      options={
        Array [
          "local",
          "development",
          "test",
          "acceptation",
          "production",
        ]
      }
    />
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <Button
      className="ml-1"
      color="secondary"
      onClick={[Function]}
    >
      Cancel
    </Button>
    <Button
      className="ml-1"
      color="primary"
      inProgress={true}
      onClick={[Function]}
    >
      Save
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`Component: OpenCloseModal ui with label: Component: OpenCloseModal => ui => with label 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  fade={true}
  isOpen={true}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  returnFocusAfterClose={true}
  role="dialog"
  scrollable={false}
  toggle={[Function]}
  unmountOnClose={true}
  zIndex={1050}
>
  <ModalHeader
    charCode={215}
    closeAriaLabel="Close"
    tag="h5"
    toggle={[Function]}
    wrapTag="div"
  >
    Choose something
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <RadioGroup
      onChange={[MockFunction]}
      optionForValue={[Function]}
      options={
        Array [
          "local",
          "development",
          "test",
          "acceptation",
          "production",
        ]
      }
    />
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <Button
      className="ml-1"
      color="secondary"
      onClick={[Function]}
    >
      Cancel
    </Button>
    <Button
      className="ml-1"
      color="primary"
      onClick={[Function]}
    >
      Save
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`Component: OpenCloseModal ui without buttons: Component: OpenCloseModal => ui => without buttons 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  fade={true}
  isOpen={false}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  returnFocusAfterClose={true}
  role="dialog"
  scrollable={false}
  toggle={[Function]}
  unmountOnClose={true}
  zIndex={1050}
>
  <ModalHeader
    charCode={215}
    closeAriaLabel="Close"
    tag="h5"
    toggle={[Function]}
    wrapTag="div"
  >
    Choose something
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <RadioGroup
      onChange={[MockFunction]}
      optionForValue={[Function]}
      options={
        Array [
          "local",
          "development",
          "test",
          "acceptation",
          "production",
        ]
      }
    />
  </ModalBody>
</Modal>
`;

exports[`Component: OpenCloseModal ui without label: Component: OpenCloseModal => ui => without label 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  fade={true}
  isOpen={false}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  returnFocusAfterClose={true}
  role="dialog"
  scrollable={false}
  toggle={[Function]}
  unmountOnClose={true}
  zIndex={1050}
>
  <ModalBody
    tag="div"
  >
    <RadioGroup
      onChange={[MockFunction]}
      optionForValue={[Function]}
      options={
        Array [
          "local",
          "development",
          "test",
          "acceptation",
          "production",
        ]
      }
    />
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <Button
      className="ml-1"
      color="secondary"
      onClick={[Function]}
    >
      Cancel
    </Button>
    <Button
      className="ml-1"
      color="primary"
      onClick={[Function]}
    >
      Save
    </Button>
  </ModalFooter>
</Modal>
`;
